File: API/Models/Messages/Message.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using API.Helpers;

namespace API.Models.Messages
{
    public class Message
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
        public string Link { get; set; }
        public int CourseId { get; set; }
        public string UserId { get; set; }
        public int RepeatAfterHour { get; set; }
        public bool IsRepeatable { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public bool SendSMS { get; set; }
        public bool SendPush { get; set; }
        public bool SendInApp { get; set; }
        public MessageType MessageType { get; set; }
        public ICollection<UserMessage> UserMessages { get; set; }

        public Message()
        {
            UserMessages = new Collection<UserMessage>();
        }
    }
}

File: Client/src/app/models/message.ts
export interface Message {
  id: number;
  title: string;
  body: string;
  link: string;
  courseId: number;
  userId: string;
  repeatAfterHour: number;
  isRepeatable: boolean;
  createdAt: Date;
  sendSMS: boolean;
  sendPush: boolean;
  sendInApp: boolean;
  messageType: number;
}
